.github/workflows/build.yml

```diff
@@ -156,29 +165,81 @@ jobs:
 
     # ntp-client
     - name: ntp-client [build]
+      if: matrix.subset == 'consensus'
       run: cabal --builddir="$CABAL_BUILDDIR" build ntp-client
 
     - name: io-sim-classes [build]
+      if: matrix.subset == 'consensus'
       run: cabal --builddir="$CABAL_BUILDDIR" build io-sim-classes
 
     - name: io-sim [build]
+      if: matrix.subset == 'consensus'
       run: cabal --builddir="$CABAL_BUILDDIR" build io-sim
 
     - name: typed-protocols [build]
+      if: matrix.subset == 'consensus'
       run: cabal --builddir="$CABAL_BUILDDIR" build typed-protocols
 
     - name: typed-protocols-examples [build]
+      if: matrix.subset == 'consensus'
       run: cabal --builddir="$CABAL_BUILDDIR" build typed-protocols-examples
 
     - name: network-mux [build]
+      if: matrix.subset == 'consensus'
       run: cabal --builddir="$CABAL_BUILDDIR" build network-mux
```

Comment:
This is all used by the network layer too. Some of these might even be network-only (`typed-protocols-examples`?), maybe @coot can confirm?